include/seqan3/alignment/matrix/detail/coordinate_matrix.hpp

```diff
@@ -0,0 +1,222 @@
+    }
+};
+
+/*!\brief The iterator for the seqan3::detail::coordinate_matrix.
+ * \ingroup alignment_matrix
+ * \implements std::forward_iterator
+ *
+ * \details
+ *
+ * Iterates over the columns of the underlying seqan3::detail::coordinate_matrix. The iterator returns a transformed
+ * std::ranges::views::iota view over the row indices. In the transformation every row index is converted to a
+ * seqan3::detail::matrix_coordinate using the seqan3::detail::convert_to_matrix_coordinate function object.
+ */
+class coordinate_matrix_iterator
+{
+private:
+
+    //!\brief The currently represented column index.
+    size_t m_column_index{0};
+    //!\brief The row index marking the end of the rows (size of one column).
+    size_t m_end_row_index{0};
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The value type.
+    using value_type = decltype(std::views::iota(static_cast<size_t>(0), static_cast<size_t>(1))
+                              | std::views::transform(convert_to_matrix_coordinate{m_column_index}));
+    //!\brief The reference type.
+    using reference = value_type;
+    //!\brief The pointer type.
+    using pointer = void;
+    //!\brief The difference type.
+    using difference_type = std::ptrdiff_t;
+    //!\brief The iterator category.
+    using iterator_category = std::forward_iterator_tag;
+    //!\}
+
+    /*!\name Constructor, assignment and destructor
+     * \{
+     */
+    coordinate_matrix_iterator() noexcept = default; //!< Defaulted.
+    coordinate_matrix_iterator(coordinate_matrix_iterator const &) noexcept = default; //!< Defaulted.
+    coordinate_matrix_iterator(coordinate_matrix_iterator &&) noexcept = default; //!< Defaulted.
+    coordinate_matrix_iterator & operator=(coordinate_matrix_iterator const &) noexcept = default; //!< Defaulted.
+    coordinate_matrix_iterator & operator=(coordinate_matrix_iterator &&) noexcept = default; //!< Defaulted.
+    ~coordinate_matrix_iterator() = default; //!< Defaulted.
+
+    /*!\brief Constructs and initialises the iterator with the current column index and the row index marking the end
+     *        of the rows (size of one column).
+     *
+     * \param[in] column_index \copybrief seqan3::detail::coordinate_matrix_iterator::m_column_index
+     * \param[in] end_row_index \copybrief seqan3::detail::coordinate_matrix_iterator::m_end_row_index
+     */
+    explicit coordinate_matrix_iterator(column_index_type<size_t> column_index,
+                                        row_index_type<size_t> end_row_index) noexcept :
+        m_column_index{column_index.get()},
+        m_end_row_index{end_row_index.get()}
+    {}
+    //!\}
+
+    /*!\name Access operators
+     * \{
+     */
+
+    //!\brief Access the pointed-to matrix coordinate column.
+    auto operator*() const
+    {
+        return std::views::iota(static_cast<size_t>(0), m_end_row_index)
```

Comment:
Within the alignment algorithm not, but it would be weird for the semantic of the class itself if it was not sized. But that does only matter if this used in some other unrelated alignment context. Not sure if that would ever be a problem. Shall I make it non-sized. I guess perforemance wise this will not make any difference.

Reply:
Oh my question meant: Can I also plug in `0u` and that would be okay?

```
return std::views::iota(0u, m_end_row_index)
```